pkg/apis/pipeline/v1beta1/customrun_types_test.go

```diff
@@ -0,0 +1,373 @@
+				Value: *v1beta1.NewStructuredValues("bar"),
+			}},
+		},
+		name: "foo",
+		want: &v1beta1.Param{
+			Name:  "foo",
+			Value: *v1beta1.NewStructuredValues("bar"),
+		},
+	}, {
+		desc: "not found",
+		spec: v1beta1.CustomRunSpec{
+			Params: []v1beta1.Param{{
+				Name:  "first",
+				Value: *v1beta1.NewStructuredValues("blah"),
+			}, {
+				Name:  "foo",
+				Value: *v1beta1.NewStructuredValues("bar"),
+			}},
+		},
+		name: "bar",
+		want: nil,
+	}, {
+		// This shouldn't happen since it's invalid, but just in
+		// case, GetParams just returns the first param it finds with
+		// the specified name.
+		desc: "multiple with same name",
+		spec: v1beta1.CustomRunSpec{
+			Params: []v1beta1.Param{{
+				Name:  "first",
+				Value: *v1beta1.NewStructuredValues("blah"),
+			}, {
+				Name:  "foo",
+				Value: *v1beta1.NewStructuredValues("bar"),
+			}, {
+				Name:  "foo",
+				Value: *v1beta1.NewStructuredValues("second bar"),
+			}},
+		},
+		name: "foo",
+		want: &v1beta1.Param{
+			Name:  "foo",
+			Value: *v1beta1.NewStructuredValues("bar"),
+		},
+	}} {
+		t.Run(c.desc, func(t *testing.T) {
+			got := c.spec.GetParam(c.name)
+			if d := cmp.Diff(c.want, got); d != "" {
+				t.Fatalf("Diff(-want,+got): %v", d)
+			}
+		})
+	}
+}
+
+func TestRunHasStarted(t *testing.T) {
+	params := []struct {
+		name          string
+		runStatus     v1beta1.CustomRunStatus
+		expectedValue bool
+	}{{
+		name:          "runWithNoStartTime",
+		runStatus:     v1beta1.CustomRunStatus{},
+		expectedValue: false,
+	}, {
+		name: "runWithStartTime",
+		runStatus: v1beta1.CustomRunStatus{
+			CustomRunStatusFields: v1beta1.CustomRunStatusFields{
+				StartTime: &metav1.Time{Time: now},
+			},
+		},
+		expectedValue: true,
+	}, {
+		name: "runWithZeroStartTime",
+		runStatus: v1beta1.CustomRunStatus{
+			CustomRunStatusFields: v1beta1.CustomRunStatusFields{
+				StartTime: &metav1.Time{},
+			},
+		},
+		expectedValue: false,
+	}}
+	for _, tc := range params {
+		t.Run(tc.name, func(t *testing.T) {
+			run := v1beta1.CustomRun{}
+			run.Status = tc.runStatus
+			if run.HasStarted() != tc.expectedValue {
+				t.Fatalf("Expected run HasStarted() to return %t but got %t", tc.expectedValue, run.HasStarted())
+			}
+		})
+	}
+}
+
+func TestRunIsDone(t *testing.T) {
+	run := v1beta1.CustomRun{
+		Status: v1beta1.CustomRunStatus{
+			Status: duckv1.Status{
+				Conditions: duckv1.Conditions{{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionFalse,
+				}},
+			},
+		},
+	}
+
+	if !run.IsDone() {
+		t.Fatal("Expected run status to be done")
+	}
+}
+
+func TestRunIsCancelled(t *testing.T) {
+	run := v1beta1.CustomRun{
+		Spec: v1beta1.CustomRunSpec{
+			Status: v1beta1.CustomRunSpecStatusCancelled,
+		},
+	}
+	if !run.IsCancelled() {
+		t.Fatal("Expected run status to be cancelled")
+	}
+	expected := ""
+	if string(run.Spec.StatusMessage) != expected {
+		t.Fatalf("Expected StatusMessage is %s but got %s", expected, run.Spec.StatusMessage)
+	}
+}
```

Comment:
I don't think we need to compare the expected msg with the run.Spec.StatusMessage here because nothing could change it in this this case, similar to TestRunIsDone.